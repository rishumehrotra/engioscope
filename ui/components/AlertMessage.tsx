import React from 'react';

const AlertMessage: React.FC<{message?: string}> = ({ message }) => (message ? (
  <div className="flex justify-center text-center items-center text-gray-600">
    <span className="text-xs rounded-full py-1 px-2 mr-2 bg-white">
      <span className="rounded-full w-3 h-3 inline-block" style={{ backgroundColor: 'rgb(255, 90, 3)' }}> </span>
      {' '}
      ALERT
    </span>
    <span className="text-sm">{message}</span>
  </div>
) : null);

export default AlertMessage;
