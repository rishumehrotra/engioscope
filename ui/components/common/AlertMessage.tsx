import React from 'react';

type AlertMessageProps = {
  message: string;
  type?: 'alert' | 'info';
}

const AlertMessage: React.FC<AlertMessageProps> = ({ message, type = 'info' }) => (message ? (
  <div className="flex justify-center text-center items-center text-gray-600">
    <span className="flex items-center text-xs rounded-full py-1 px-2 mr-2 bg-white">
      <span
        className={`rounded-full w-3 h-3 inline-block ${type === 'alert' ? 'bg-red-400' : 'bg-purple-500'}`}
      />
      <span className="ml-1">
        {type.toUpperCase()}
      </span>
    </span>
    <span className="text-sm">{message}</span>
  </div>
) : null);

export default AlertMessage;
