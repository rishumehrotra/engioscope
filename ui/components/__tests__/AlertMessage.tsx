import React from 'react';
import { render } from '@testing-library/react';
import AlertMessage from '../AlertMessage';

describe('Alert Message', () => {
  it('should render the message passed', () => {
    const message = 'This is an alert message';
    const alert = render(<AlertMessage message={message} />);
    expect(alert.findByText(message)).toBeTruthy();
  });

  it('should render an alert message with an alert pill and appropriate styles', () => {
    const { asFragment } = render(<AlertMessage message="This is an alert message" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
