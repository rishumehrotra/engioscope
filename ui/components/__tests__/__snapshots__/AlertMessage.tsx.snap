// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Message > should render an alert message with an alert pill and appropriate styles 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center text-center items-center text-gray-600"
  >
    <span
      class="flex items-center text-xs rounded-full py-1 px-2 mr-2 bg-white"
    >
      <span
        class="rounded-full w-3 h-3 inline-block bg-purple-500"
      />
      <span
        class="ml-1"
      >
        INFO
      </span>
    </span>
    <span
      class="text-sm"
    >
      This is an alert message
    </span>
  </div>
</DocumentFragment>
`;
